externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/ExecuteStatement.scala

```diff
@@ -159,6 +160,25 @@ class ExecuteStatement(
         override def iterator: Iterator[Any] = incrementalCollectResult(resultDF)
       })
     } else {
+      val sparkSave = spark.conf.getOption(OPERATION_RESULT_SAVE_TO_FILE.key).map(_.toBoolean)
+        .getOrElse(session.sessionManager.getConf.get(OPERATION_RESULT_SAVE_TO_FILE))
+      lazy val threshold =
+        session.sessionManager.getConf.get(OPERATION_RESULT_SAVE_TO_FILE_THRESHOLD)
+      if (hasResultSet && sparkSave && shouldSaveResultToHdfs(resultMaxRows, threshold, result)) {
+        val sessionId = session.handle.identifier.toString
+        val savePath = session.sessionManager.getConf.get(OPERATION_RESULT_SAVE_TO_FILE_PATH)
+        val fileName = s"$savePath/$engineId/$sessionId/$statementId"
+        val colName = range(0, result.schema.size).map(x => "col" + x)
+        if (resultMaxRows > 0) {
+          result.toDF(colName: _*).limit(resultMaxRows).write
```

Comment:
Use `resultDF` instead of `result`. Also, is `toDF(colName: _*)` necessary?

Reply:
If the result has duplicate columns, we can not write it to file, so we rename all col name to avoid this case
```
spark.sql("select 1 as a,2 as a").write("/filepath")

org.apache.spark.sql.AnalysisException: Found duplicate column(s) when inserting into hdfs
```

Reply:
@lsm1 let's add such information to the comments

Reply:
there is another known issue as I mentioned in the issue comment

> directly call `df.write` will introduce an extra shuffle for the outermost limit, and hurt performance

I think we should also add this known issue to the comment and create a new ticket to track this issue.